### ISSUE TITLE
[BUG][blob csi] blob is not mounted at pod starting.

### ISSUE THREAD

**Issue by yulin-li** [Azure/AKS#4112]
**Describe the bug**

**To Reproduce**
- install blob csi extension
- create pv and pvc then mount to a simple pod
- list the mount path, e.g.,
```
 # list the file every 0.1 seconds
        while true; do
          ls /mnt | jq -R -s -c 'split("\n")[:-1]'
          sleep 0.1
        done
```
- the result is empty at beginning and then got the correct files, sample outptus
```
[]
[]
[]
[]
[]
...
['a.file']
```

**Expected behavior**
The pod should not start until mount is ready

**Environment (please complete the following information):**
 - Kubernetes version aks 1.26

**Additional context**

blob fuse mount options
```
mountOptions:
    - '-o allow_other'
    - '--file-cache-timeout-in-seconds=1200'
 ```

**Comment by andyzhangx** [comment]
@yulin-li what's the blob csi driver version you are using? are you using the managed blob csi driver on AKS?

**Comment by andyzhangx** [comment]
btw, when do you find those files exist after blobfuse mount, e.g. 1s or so?

**Comment by yulin-li** [comment]
yes, it's managed blob csi so it should be the default version of aks 1.26

The latency is about 3-5s from my observations

**Comment by andyzhangx** [comment]
@vibhansa-msft is this expected behavior of blobfuse mount? currently we could see that after blobfuse mount successfully, the file in storage account would exist after 3-5s

**Comment by andyzhangx** [comment]
@yulin-li have you set `--cancel-list-on-mount-seconds=1s` to make refresh faster after mount?

**Comment by andyzhangx** [comment]
@yulin-li sorry after second look, actually our blob csi driver would set as 10s by default, so if you set `--cancel-list-on-mount-seconds=0`, it would overwrite existing mount option, thus there should no waiting:
```
		// prevent billing charges on mounting
		"--cancel-list-on-mount-seconds": "10",
```

**Comment by andyzhangx** [comment]
this is by design since we have set `cancel-list-on-mount-seconds` as 10s to prevent billing charges on mounting, just override this option if you want to make it quicker
